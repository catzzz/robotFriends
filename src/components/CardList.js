import React, { Component } from 'react'

import Card from './Card';
export default class CardList extends Component {
    

    render() {
        console.log("CardList render")
        const { robots } = this.props;
        const cardComponent = robots.map((robot,index) => {
            return <Card name ={robot.name} email={robot.email} id={robot.id} key={robot.id}/>
        });

        return (
            <div>
                {cardComponent}
            </div>
        )
    }
}
