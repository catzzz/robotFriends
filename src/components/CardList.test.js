import CardList from './CardList';
import { create } from "react-test-renderer";
import React from 'react';

it('expect to render CardList componet ',()=>{
    const mockRobots = [
        {
            id:1,
            name:'Jimmy',
            username: 'John John',
            email:'jimmy@gmail.com'
        }
    ]
    const app = create(<CardList robots = {mockRobots}/>);
    expect(app.toJSON()).toMatchSnapshot();
})
