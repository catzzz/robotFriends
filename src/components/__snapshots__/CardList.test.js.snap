// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList componet  1`] = `
<div>
  <div
    className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <h1>
      Robots
    </h1>
    <img
      alt="robots"
      src="https://robohash.org/1?200x200"
    />
    <div>
      <h2>
        Jimmy
      </h2>
      <p>
         
        jimmy@gmail.com
      </p>
    </div>
  </div>
</div>
`;
